src/main/java/sirius/biz/util/SevenZipAdapter.java

```diff
@@ -82,13 +83,16 @@ public ISequentialOutStream getStream(int index, ExtractAskMode extractAskMode)
         }
 
         currentFilePath = fetchFilePath(index);
-        currentLastModified = Instant.ofEpochSecond((Long) inArchive.getProperty(index, PropID.LAST_MODIFICATION_TIME));
 
         // Enforce the filename filter...
         if (!filter.test(currentFilePath)) {
             return null;
         }
 
+        Date lastModificationTime = (Date) inArchive.getProperty(index, PropID.LAST_MODIFICATION_TIME);
+        currentLastModified =
+                lastModificationTime == null ? Instant.now() : Instant.ofEpochSecond(lastModificationTime.getTime());
```

Comment:
`Optional.ofNullable(lastModificationTime).map(date -> Instant.ofEpochSecond(date.getTime())).orElse(Instant.now());`

typing it out i realize your version is probably cleaner..

Reply:
I think its cleaner with optional chain as then we don't have two similiar named date variables in scope.

Reply:
Wollts auch fast vorschlagen :D